### PR TITLE
[CD] Add pytorch xpu wheel build in nightly

### PR DESCRIPTION
**Pull request by chuanqi129**
Add pytorch xpu wheel build in nightly after the xpu build image enabling PR https://github.com/pytorch/builder/pull/1879 merged. Link to #114850

### CODE REVIEW

**Comment by chuanqi129** [comment]
Triton build failures caused by https://github.com/triton-lang/triton/issues/4209, waiting triton team to fix

File: .github/scripts/generate_binary_build_matrix.py [hunk 1]
```diff
@@ -24,6 +25,7 @@
 
 ROCM_ARCHES = ["6.0", "6.1"]
 
+XPU_ARCHES = ["xpu"]
```
Comment (atalman) [inline .github/scripts/generate_binary_build_matrix.py:28]: @chuanqi129 separate this PR into 2.
-  First build triton-xpu wheel. The changes in this file and [generated-linux-binary-manywheel-nightly.yml](https://github.com/pytorch/pytorch/pull/129560/files#diff-7296837255a68baadd9f386ec756ab581773111df14bc37752d68f53c7b36d79) are not required at this point
- Second PR actually add manywheel build.

Comment (chuanqi129) [reply .github/scripts/generate_binary_build_matrix.py:28]: Thanks @atalman. Make sense, have split the triton-xpu wheel build in PR #129730.

**Comment by chuanqi129** [comment]
The xpu 3.8 wheel build failed on numpy 1.15 installation should be fixed by PR https://github.com/pytorch/builder/pull/1909

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml [hunk 2]
```diff
@@ -751,6 +751,118 @@ jobs:
...
+  manywheel-py3_8-xpu-build:
+    if: ${{ github.repository_owner == 'pytorch' }}
+    uses: ./.github/workflows/_binary-build-linux.yml
+    with:
+      PYTORCH_ROOT: /pytorch
+      BUILDER_ROOT: /builder
+      PACKAGE_TYPE: manywheel
+      # TODO: This is a legacy variable that we eventually want to get rid of in
+      #       favor of GPU_ARCH_VERSION
+      DESIRED_CUDA: xpu
```
Comment (EikanWang) [inline .github/workflows/generated-linux-binary-manywheel-nightly.yml:763]: What's the meaning of `DESIRED_CUDA`?

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml [hunk 3]
```diff
@@ -1465,6 +1577,118 @@ jobs:
       conda-pytorchbot-token-test: ${{ secrets.CONDA_PYTORCHBOT_TOKEN_TEST }}
     uses: ./.github/workflows/_binary-upload.yml
 
+  manywheel-py3_9-xpu-build:
```
Comment (EikanWang) [inline .github/workflows/generated-linux-binary-manywheel-nightly.yml:1580]: Most of the code is copy-past. Compared with Py3.8, is the python version the only difference? I'm just wondering if we can reuse the code (https://yaml.org/spec/1.2.2/#rule-c-ns-anchor-property)

Comment (chuanqi129) [reply .github/workflows/generated-linux-binary-manywheel-nightly.yml:1580]: Hi @EikanWang , this workflow is generated by script `.github/scripts/generate_ci_workflows.py` based on template [.github/templates/linux_binary_build_workflow.yml.j2](https://github.com/pytorch/pytorch/blob/4da45c5243a7be402723660c4e58cf0249411323/.github/templates/linux_binary_build_workflow.yml.j2)

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml (hunk 2 above)
Comment (chuanqi129) [reply .github/workflows/generated-linux-binary-manywheel-nightly.yml:763]: Same here. And the `DESIRED_CUDA` shared with all devices, including `cpu`, `rocm` etc.

**Comment by chuanqi129** [comment]
The build crash will be fixed by PR https://github.com/pytorch/pytorch/pull/130333

**Comment by chuanqi129** [comment]
Hi @atalman , all PR changes related test jobs are passed, especially for the XPU manylinux wheel build and test jobs. But there are some unrelated test job failures caused by conda, could you please help to double check and review the PR again? Thanks

**Review by atalman — APPROVED** [review]

File: .circleci/scripts/binary_linux_test.sh [hunk 4]
```diff
@@ -118,6 +118,12 @@ if [[ "$PACKAGE_TYPE" == libtorch ]]; then
   cd /tmp/libtorch
 fi
 
+if [[ "$GPU_ARCH_TYPE" == xpu ]]; then
+  # Workaround for __mkl_tmp_MOD unbound variable issue
```
Comment (atalman) [inline .circleci/scripts/binary_linux_test.sh:122]: Can you please add a link to the issue here

Comment (chuanqi129) [reply .circleci/scripts/binary_linux_test.sh]: Added

**Pull request closed without merging**